jsplot/src/main/java/tech/tablesaw/plotly/api/BubblePlot.java

```diff
@@ -38,22 +42,43 @@ public static Figure create(
   }
 
   public static Figure create(
-      String title, Table table, String xCol, String yCol, String sizeColumn) {
-    Layout layout = Layout.builder(title, xCol, yCol).build();
-
-    Marker marker =
-        Marker.builder()
-            .size(table.numberColumn(sizeColumn))
-            // .opacity(.75)
-            .build();
-
-    ScatterTrace trace =
-        ScatterTrace.builder(table.numberColumn(xCol), table.numberColumn(yCol))
-            .marker(marker)
-            .build();
+      String title,
+      Column xColumn,
+      Column yColumn,
+      NumericColumn sizeColumn,
+      double[] color,
+      SizeMode sizeMode,
+      Double opacity) {
+
+    Layout layout = Layout.builder(title, xColumn.name(), yColumn.name()).build();
+
+    Marker marker = null;
+    MarkerBuilder builder = Marker.builder();
+    if (sizeColumn != null) {
+      builder.size(sizeColumn);
+    }
+    if (opacity != null) {
+      builder.opacity(opacity);
+    }
+    if (color != null) {
+      builder.color(color);
+    }
+    if (sizeMode != null) {
+      builder.sizeMode(sizeMode);
+    }
+    marker = builder.build();
+
+    ScatterTrace trace = ScatterTrace.builder(xColumn, yColumn).marker(marker).build();
     return new Figure(layout, trace);
   }
 
+  public static Figure create(String title, Table table, String xCol, String yCol, String sizeCol) {
+    NumericColumn xColumn = table.numberColumn(xCol);
+    NumericColumn yColumn = table.numberColumn(yCol);
+    NumericColumn sizeColumn = table.numberColumn(sizeCol);
```

Comment:
... and actually here sizeCol could be allowed to be null and just init sizeColumn to null in that case.